src/objective-c/GRPCClient/GRPCCall.h

```diff
@@ -155,6 +155,19 @@ typedef NS_ENUM(NSUInteger, GRPCErrorCode) {
 };
 
 /**
+ * Flags for options of a gRPC call
+ *
+ */
+typedef NS_OPTIONS(NSUInteger, GRPCCallFlags) {
+  /** Signal that the call is idempotent */
+  GRPCFlagIdempotentRequest = 0x00000010,
+  /** Signal that the call should not return UNAVAILABLE before it has started */
+  GRPCFlagIgnoreConnectivity = 0x00000020,
+  /** Signal that the call is cacheable. GRPC is free to use GET verb */
+  GRPCFlagCacheableRequest = 0x00000040,
```

Comment:
As cacheable implies idempotent, we should only allow 3 values (default, idempotent, cacheable), rather than the 4 combinations that two flags provide.